项目设置和小游戏分包等修改 (#1664)
cocos-creator/creator-docs: en/editor/project/index.md

**Project Settings** consists of several different subpages, including **Macro Config**, **Feature Cropping**, **Project Data**, **Layers**, **Physics**, **Scripts**, and **Texture Compression**. The **Project Settings** panel will automatically save the changes after modifying the settings.

## Macro Config

**Macro Config** provides a shortcut to modify the macro configuration. The configured macros will take effect on preview and build, and will also follow the custom engine configuration to update the default values of the current macro configuration.

![macro](./index/macro.png)

- **ENABLE_TILEDMAP_CULLING**: whether or not to enable TiledMap auto-culling. It's enabled by default. Note that if `skew` and `rotation` are set in the TiledMap, it's necessary to turn it off manually, otherwise it will cause rendering errors.

> - **TOUCH_TIMEOUT**: the duration of the delay used to screen a contact object to see if it has failed and can be removed. This value can be modified to get the effect you need, the default value is 5000 ms.

Reviewer: ```suggestion
- **TOUCH_TIMEOUT**: the duration of the delay used to screen a contact object to see if it has failed and can be removed. This value can be modified to get the effect you need, the default value is 5000 ms. Please refer to the API [TOUCH_TIMEOUT](__APIDOC__/en/modules/core.html#macro-1.touch_timeout) documentation for details.
```

- **ENABLE_TRANSPARENT_CANVAS**: used to set whether the Canvas background supports alpha channels.

    - If enabled, the Canvas background will be transparent and show the other DOM elements behind it.
    - If disabled, it will result in higher performance.

- **ENABLE_WEBGL_ANTIALIAS**: used to set whether to turn on the anti-aliasing option when creating WebGL Context. It's enabled by default.

- **CLEANUP_IMAGE_CACHE**: whether to clear the image cache after uploading a texture to GPU. If the cache is cleared, [Dynamic Atlas](../../advanced-topics/dynamic-atlas.md) will not be supported. It's disabled by default.

- **ENABLE_MULTI_TOUCH**: whether to enable multi-touch.

> - **MAX_LABLE_CANVAS_POOL_SIZE**: the maximum number of Canvas objects used by the Label. Please adjust it according to the number of Labels in the same scene of the project.

Reviewer: ```suggestion
- **MAX_LABLE_CANVAS_POOL_SIZE**: set the maximum number of Canvas object pools used by the Label, and adjust it according to the number of Labels in the same scene of the project.
```

## Macro Config

**Macro Config** provides a shortcut to modify the macro configuration, the configured macro will take effect when previewing, building The configured macro will take effect on preview, build, and will also follow the custom engine configuration to update the default values of the current macro configuration.

> - **ENABLE_TILEDMAP_CULLING**: whether or not enabled tiled map auto culling. If you set the TiledMap skew or rotation, then need to manually disable this, otherwise, the rendering will be wrong.

Reviewer 2: ```suggestion
- **ENABLE_TILEDMAP_CULLING**: whether or not tiled map auto culling is enabled. If setting the TiledMap skew or rotation, then it is necessary to manually disable this setting, otherwise, the rendering will be wrong.
```

> - **TOUCH_TIMEOUT**: the timeout to determine whether a touch is no longer active and should be removed. You can modify this value to have a better behavior if you find it's not enough.

Reviewer 2: ```suggestion
- **TOUCH_TIMEOUT**: the timeout to determine whether a touch is no longer active and should be removed. This value can be modified to have a better behavior if necessary.
```

- **ENABLE_TRANSPARENT_CANVAS**: boolean that indicates if the canvas contains an alpha channel, default sets to false for better performance.

- **ENABLE_WEBGL_ANTIALIAS**: boolean that indicates if the WebGL context is created with `antialias` option turned on, default value is false.

> - **CLEANUP_IMAGE_CACHE**: whether to clear the original image cache after uploaded a texture to GPU. If cleared, [Dynamic Atlas](https://docs.cocos.com/creator/manual/en/advanced-topics/dynamic-atlas.html) will not be supported.

Reviewer: 用相对路径

- **ENABLE_MULTI_TOUCH**: whether to enable multi-touch.

- **MAX_LABLE_CANVAS_POOL_SIZE**: the maximum size of the canvas pool used by Label, please adjust according to the number of label component in the same scene of the project.

![macro](./index/macro.png)

> For more specific information and code about the engine macro module, you can refer to **Engine Macro** ([GitHub](https://github.com/cocos-creator/engine/blob/3d/cocos/core/platform/macro.ts#L824).

Reviewer 2: ```suggestion
For more specific information and code about the engine macro module, please refer to the [**Engine Macro** source](https://github.com/cocos-creator/engine/blob/3d/cocos/core/platform/macro.ts#L824).
```

> For more specific information and code about the engine macro module, please refer to the [**Engine Macro** source](https://github.com/cocos-creator/engine/blob/3d/cocos/core/platform/macro.ts#L824).

Reviewer: ```suggestion
For more specific information and code about the engine macro module, please refer to the [Engine Macro source](https://github.com/cocos-creator/engine/blob/3d/cocos/core/platform/macro.ts#L824).
```

> For specific information and code about the engine macro module, you can refer to **Engine Macro** ([GitHub](https://github.com/cocos-creator/engine/blob/3d/cocos/core/platform/macro.ts#L824) | [Gitee](https://gitee.com/mirrors_cocos-creator/engine/blob/3d/cocos/core/platform/macro.ts#L824)).

Reviewer: 移除 gitee 的链接

![scripting](./index/scripting.png)

> - **Conforming class fields**: if true, class fields are implemented using `Define` semantics; otherwise, class fields are implemented using `Set` semantics.

Reviewer: if true -> if this option is enabled,

![scripting](./index/scripting.png)

- **Conforming class fields**: if enabled, class fields are implemented using `Define` semantics; otherwise, class fields are implemented using `Set` semantics.

> - **Allows declaration of class fields**: if enabled, `declare` keyword may be used to declare class fields. In such case, class fields without explicit initializer would be initialized to undefined, according to the specification.

Reviewer: ```suggestion
- **Allows declaration of class fields**: if enabled, the `declare` keyword is allowed to declare class fields in TypeScript scripts. When a field is declared with `declare` and no explicit initialization is specified, it will be initialized to `undefined` according to the specification.
```

> - **Enable relaxed mode**: enable "loose" compilation mode.

Reviewer: ```suggestion
- **Enable relaxed mode**: if enabled, the script will be compiled in non-strict mode.
```
